p4-16/psa/psa.p4

```diff
@@ -55,17 +54,19 @@ typedef bit<unspecified> PacketLength_t;
 typedef bit<unspecified> EgressInstance_t;
 typedef bit<unspecified> Timestamp_t;
 
+const   PortId_t         PORT_RECIRCULATE = unspecified;
 const   PortId_t         PORT_CPU = unspecified;
 // END:Type_defns
 
-// typedef bit<unspecified> InstanceType_t;
-// const   InstanceType_t   INSTANCE_NORMAL = unspecified;
 #endif
 
 // BEGIN:Metadata_types
 enum InstanceType_t {
-    NORMAL,     /// Packet is "normal", i.e. none of the other cases below
-    CLONE,      /// Packet was created via a clone operation
+    NORMAL,     /// Packet received by ingress that is none of the cases below.
+    NORMAL_UNICAST,   /// Packet received by egress that is none of the cases below, and unicast
+    NORMAL_MULTICAST, /// Packet received by egress that is none of the cases below, and multicast
+    CLONE_I2E,  /// Packet created via a clone operation in ingress, destined for egress
+    CLONE_E2E,  /// Packet created via a clone operation in egress, destined for egress
```

Comment:
Having all of these values makes distinguishing these cases at the beginning of ingress or egress processing possible.